tests/tools/test_dist_part.py

```diff
@@ -209,15 +213,16 @@ def read_orig_ids(fname):
             )
 
 
-@pytest.mark.parametrize("num_chunks", [1, 3, 4, 8])
-@pytest.mark.parametrize("num_parts", [1, 3, 4, 8])
-def test_pipeline_basics(num_chunks, num_parts):
-    _test_pipeline(num_chunks, num_parts)
+@pytest.mark.parametrize("num_chunks", [1, 2, 3, 4, 8, 10])
```

Comment:
do we really need to test so many combinations? adding a new parameter will generate X times testcases. And each testcase takes non-trivial seconds to finish:
```
17.06s call     tests/tools/test_dist_part.py::test_pipeline_basics[4-8-8]
13.93s call     tests/tools/test_dist_part.py::test_pipeline_basics[2-8-10]
13.30s call     tests/tools/test_dist_part.py::test_pipeline_basics[3-6-10]
13.11s call     tests/tools/test_dist_part.py::test_pipeline_basics[2-8-8]
13.04s call     tests/tools/test_dist_part.py::test_pipeline_basics[3-6-8]
13.02s call     tests/tools/test_dist_part.py::test_pipeline_basics[4-8-10]
10.63s call     tests/tools/test_dist_part.py::test_pipeline_basics[4-4-8]
10.17s call     tests/tools/test_dist_part.py::test_pipeline_basics[4-4-10]
9.77s call     tests/tools/test_dist_part.py::test_pipeline_basics[4-4-4]
9.15s call     tests/tools/test_dist_part.py::test_pipeline_basics[2-6-8]
8.91s call     tests/tools/test_dist_part.py::test_pipeline_basics[3-3-8]
8.80s call     tests/tools/test_dist_part.py::test_pipeline_basics[3-3-3]
8.70s call     tests/tools/test_dist_part.py::test_pipeline_basics[3-3-4]
8.46s call     tests/tools/test_dist_part.py::test_pipeline_basics[2-4-8]
8.36s call     tests/tools/test_dist_part.py::test_pipeline_basics[2-6-10]
```

Reply:
Agreed, I will reduce some test cases so that total time to run the CI tests will reduce a little bit.